addons/block_code/serialization/block_name_tree_node.gd

```diff
@@ -0,0 +1,13 @@
+extends Resource
+
+const BlockNameTreeNode = preload("res://addons/block_code/serialization/block_name_tree_node.gd")
+
+@export var name: StringName
+@export var children: Array[BlockNameTreeNode]
```

Comment:
If I understand correctly, this is [path_child_pairs](https://github.com/endlessm/godot-block-coding/blob/55acc2d917eb1250c81d92262f6e7d14a4df45b0/addons/block_code/serialization/block_serialization.gd#L6) in the current serialization. And that allowed to map multiple snap points to children blocks. I think that the new format should also support that. This is why an if/else block is not possible anymore. Ideally this should be a Dictonary that maps slot names to children blocks, so no need to have NodePaths in the serialization like currently, eg from the Pong serialization:
```
path_child_pairs = [[NodePath("VBoxContainer/SnapPoint"), SubResource("Resource_ftveg")]]
```
For example:
- An entry block like "On ready" can have a single slot named "body". So the Dictionary would be: `{ "body": [BlockSerialization, BlockSerialization]}`
- A sentence block like "Print" has no slots, so the Dictionary is empty: `{}`
- An if/else block has 2 bodies: `{"if_body": [BlockSerialization], "else_body": [BlockSerialization, BlockSerialization]}`

Do you think this would be too difficult to change at this point?

Reply:
Correct, I like the idea of having a dictionary with snap point slot names as keys, and a list of blocks as values.

I don't think it would be too difficult to change if you wanted to do it in this PR. You would have to probably make a subclass of `BlockDefinition` like `MultiBlockDefinition` that has an array of `code_template`s and `display_template`s. Then, when traversing the tree to generate the other tree states, you would have to traverse the children in each slot, as well as all of the arguments for the slot headers (which may be a bit trickier).

I still think it would probably be better to just iterate on this later since we are still a little unsure (personally I found the Scratch-type if/else blocks a little annoying sometimes), and we aren't really losing any functionality here, just some safety checking. And there's plenty to look at for safety checking in the future :grin:. I think the main benefit of getting this PR in will be to separate the states. If you want to reimplement control blocks it will be much less hacky than it was before.